src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs

```diff
@@ -79,27 +79,66 @@ public byte[] SignData(byte[] data, HashAlgorithmName hashAlgorithm)
             {
                 throw new ArgumentNullException(nameof(data));
             }
+
             return SignData(data, 0, data.Length, hashAlgorithm);
         }
 
+        public byte[] SignData(byte[] data, HashAlgorithmName hashAlgorithm, DSASignatureFormat signatureFormat)
+        {
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
```

Comment:
You should also check that signatureFormat is one of the known inputs.  Especially since the enum and this type are in the same assembly, they can't skew.